homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,281 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_COMMAND_OFF,
+    CONF_COMMAND_ON,
+    CONF_DEVICE,
+    CONF_DEVICE_ID,
+    CONF_DEVICES,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import (
+    async_entries_for_config_entry,
+    async_get_registry,
+)
 
-from . import DOMAIN
+from . import DOMAIN, get_device_id, get_rfx_object
+from .binary_sensor import supported as binary_supported
+from .const import (
+    CONF_AUTOMATIC_ADD,
+    CONF_DATA_BITS,
+    CONF_DEBUG,
+    CONF_FIRE_EVENT,
+    CONF_OFF_DELAY,
+    CONF_SIGNAL_REPETITIONS,
+    DEVICE_PACKET_TYPE_LIGHTING4,
+)
+from .cover import supported as cover_supported
+from .light import supported as light_supported
+from .switch import supported as switch_supported
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_OFF_DELAY_ENABLED = CONF_OFF_DELAY + "_enabled"
+CONF_REMOVE_DEVICE = "remove_device"
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle Rfxtrx options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize rfxtrx options flow."""
+        self._config_entry = config_entry
+        self._global_options = None
+        self._selected_device = None
+        self._selected_device_event_code = None
+        self._rfxobj = None
+        self._device_entries = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        return await self.async_step_prompt_options()
+
+    async def async_step_prompt_options(self, user_input=None):
+        """Prompt for options."""
+        errors = {}
+
+        if user_input is not None:
+            self._global_options = {
+                CONF_DEBUG: user_input[CONF_DEBUG],
+                CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+            }
+            if CONF_DEVICE in user_input:
+                device_data = self._get_device_data(user_input[CONF_DEVICE])
+                if not device_data.get("event_code"):
+                    errors = {"base": "unknown_event_code"}
+                else:
+                    event_code = device_data["event_code"]
+                    self._selected_device_event_code = event_code
+                    self._selected_device = self._config_entry.data[CONF_DEVICES][
+                        event_code
+                    ]
+                    return await self.async_step_set_device_options()
+            if CONF_REMOVE_DEVICE in user_input:
+                device_data = self._get_device_data(user_input[CONF_REMOVE_DEVICE])
+                if not device_data.get("event_code"):
+                    errors = {"base": "unknown_event_code"}
+                else:
+                    event_code = device_data["event_code"]
+                    device_id = device_data["device_id"]
+                    self.hass.helpers.dispatcher.async_dispatcher_send(
+                        f"{DOMAIN}_{device_id}"
```

Comment:
this is a very non descriptive event. Need a const value that describe what it is.